### PR TITLE
OCPBUGS-18545: CORS-2445: GCP: Add osImage to the install config

### PR DESCRIPTION
**Pull request by barbacbd**
- The initial stage towards enabling users to specify a boot images from Google Cloud Marketplace. This involves the introduction of a new field called oSImage. Within this field, users can indicate the project and name of the desired image.
- Fixup tests and added generated data.
- Updated google compute version to the minimum version where Architecture is a valid field in the Image struct. The field is required for GCP marketplace images.

### CODE REVIEW

**Comment by patrickdillon** [comment]
> /label backport-risk-assessed

Brent mentioned in slack that some further changes may be needed here, although it looks like this is testing well. If further changes are needed, let's re-evaluate this label.

File: data/data/install.openshift.io_installconfigs.yaml [hunk 1]
```diff
@@ -375,6 +375,26 @@ spec:
...
+                              type: string
+                            project:
+                              description: Project defines the name of the project
+                                containing the image.
+                              type: string
+                          required:
+                          - name
+                          - project
+                          type: object
+                        tags:
```
Comment (mjpytlak) [inline data/data/install.openshift.io_installconfigs.yaml:392]: @barbacbd Looks like you are introducing user specified tags here and on lines 1047 and 2238. Is that intentional? We released that enhancement in 4.12. https://issues.redhat.com/browse/CORS-2209

**Comment by jianli-wei** [comment]
/hold

Please refer to [comment](https://issues.redhat.com/browse/OCPBUGS-18545?focusedId=23006950&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23006950), thanks!

**Comment by barbacbd** [comment]
@jianli-wei I pulled that data out of the PR (cherry-pick issues). Looks like it should mirror previous backports except for the dependency updates.

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Pull request merged**